pkg/log/fields.go

```diff
@@ -0,0 +1,91 @@
+package log
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+)
+
+func mergeFields(receiver map[string]interface{}, input map[string]interface{}) map[string]interface{} {
+	newMap := make(map[string]interface{}, len(receiver)+len(input))
+
+	for k, v := range receiver {
+		newMap[k] = prepareForLog(v)
+	}
+
+	for k, v := range input {
+		newMap[k] = prepareForLog(v)
+	}
+
+	return newMap
+}
+
+func prepareForLog(v interface{}) interface{} {
+	switch t := v.(type) {
+	case error:
```

Comment:
Does this also work with the multi-error package?

Reply:
As the multi-error implements the `error` interface it should work, yes..  
And this code is just copied from the old `mon` package, so it is somewhat battle tested already.. :grimacing: